Tools/TestWebKitAPI/cocoa/TestWKWebView.mm

```diff
@@ -524,6 +524,13 @@ - (void)_setUpTestWindow:(NSRect)frame
 #endif
 }
 
+- (void)setUpAndAddToTestWindow
+{
+    ASSERT(!_hostWindow);
+    [self _setUpTestWindow:self.frame];
+    [self addToTestWindow];
```

Comment:
Seems _setUpTestWindow already does what addToTestWindow does.

Also, why do we need this new function at all. Why not pass `addToWindow:YES` when constructing the TestWKWebView? (which would call the internal `_setUpTestWindow` for you).

Reply:
If we use 'addToWindow:YES' at construction time, it creates the window and adds the view to it before we have a chance to set the UI delegate. This causes the first-use logic to fire before we have a chance to register our own handler.

Reply:
I see. Then maybe we could have the tests call `[webView addToTestWindow]` and update `addToTestWindow` to call `_setUpTestWindow` internally if `_hostWindow` is nil.

Reply:
I have a better approach that will be less confusing. I'll update the patch.